neo/core/analogsignal.py

```diff
@@ -265,7 +265,8 @@ def __getitem__(self, i):
                     raise TypeError("%s not supported" % type(j))
                 if isinstance(k, (int, np.integer)):
                     obj = obj.reshape(-1, 1)
-                obj.array_annotate(**deepcopy(self.array_annotations_at_index(k)))
+                if k is not None:  # matplotlib _check_1d() calls__getitem__ with (:, None)
+                    obj.array_annotate(**deepcopy(self.array_annotations_at_index(k)))
```

Comment:
Shouldn't the array_annotation for the case `k is None` still be added up to the last index available?

Reply:
On reflection, `None` shouldn't be allowed at all. `x[:, None]` adds an empty dimension to a NumPy array `x`. This doesn't make sense for `AnalogSignal`, which can only be 2D, so I've updated the PR to have Neo raise an IndexError (which also solves the original problem with matplotlib).

The alternative would be to cast the object to a 3D Quantities array, but I think that would be unexpected behaviour.

Reply:
I think a lot of the `AnalogSignal` code is written with 2 or more dimensions in mind, only assuming the first and the last dimension have a fixed meaning. So maybe adding a new dimension in case of `[ : , None]` would be the cleaner way? @samuelgarcia Any strong opinion on that?

Reply:
According to the AnalogSignal documentation these have to be 2D. So we should rather raise an error than adding a new dimension.